### PR TITLE
[Chore] Docker image Based Deployment

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

Introduce docker image based deployment.

## Type of change

- [x] Chore.

## Screenshots

N/A

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,25 +1,59 @@
-name: Dokku CD
+name: main
 on:
   push:
     branches: [main]
 env:
-  DOKKU_APP_NAME: "outbreak"
-  DOKKU_HOST: "dokku-1.hurumap.org"
   DOKKU_REMOTE_BRANCH: "master"
+  DOKKU_REMOTE_URL: "ssh://dokku@dokku-1.hurumap.org/outbreak"
```
Comment (esirK) [inline .github/workflows/main.yml:7]: I think we can make this configurable instead of hardcoding the URL.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,25 +1,81 @@
...
-          git-push-flags: ${{ env.GIT_PUSH_FLAGS }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+
+      - name: Check if version is bumped
+        id: version-check
+        uses: EndBug/version-check@v1
```
Comment (esirK) [inline .github/workflows/main.yml:30]: Not sure I fully understand how this `version-check` is working for us. 
Are we just building new images if the `package.json` shows a change?
And how are we reflecting this to the Dokku `Dockerfile`?

Comment (kilemensi) [reply .github/workflows/main.yml:30]: package version is stored in [`package.json`](https://github.com/CodeForAfrica/outbreak.AFRICA/pull/143/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R102). [`version-check`](https://github.com/EndBug/version-check#usage) checks if that version has actually been changed. You can change the file without making a new version e.g. when you want to make 2 quick fixes before deploying, etc.

Once there is a new version is detected that's when a new docker image will be built using the new version as tag (docker tag === project version).

Changing this version is done manually by whoever needs to deploy to prod. (there are tools like `yarn version` to bump `package.json` version but we'll need to find one that will bump on `Dockerfile` as well).

Comment (esirK) [reply .github/workflows/main.yml:30]: Aaah makes more sense now.

File: .github/workflows/main.yml (hunk 1 above)
Comment (esirK) [reply .github/workflows/main.yml:7]: @kilemensi  I think using `${{ env.DOKKU_REMOTE_URL }}` as used below should also work here.

**Review by esirK — APPROVED** [review]
LGTM

File: .github/workflows/main.yml (hunk 1 above)
Comment (kilemensi) [reply .github/workflows/main.yml:7]: Not sure I get you @esirK, I'm _declaring_ the env here. Or do you mean we should use GitHub secret for this?

**Pull request merged**